### DISCUSSION TITLE
[6 luglio 2023] Complessità

### DISCUSSION THREAD

**Discussion by matypist** [sapienzastudentsnetwork/automi-calcolabilita-complessita#15] [Complessità]
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/69389461/b08fec5c-e355-494a-97e7-548bb4ee7d05)

**Comment by aflaag** [comment]
Poiche $SAT \in \textsf{NP-Complete}$ per il teorema di Cook-Levin, per definizione si ha che $$SAT \in \textsf{NP-Complete} \implies SAT \in \textsf{NP} \iff \overline{SAT} \in \textsf{coNP}$$ inoltre, per $\textsf{NP}$-difficoltà di $SAT$, si ha che $$\forall L \in \textsf{NP} \quad L \leq_\mathrm P SAT \iff \overline L \leq_\mathrm P \overline{SAT}$$ e poiché $L \in \textsf{NP} \iff \overline L \in \textsf{coNP}$ per definizione, segue che ogni linguaggio in $\textsf{coNP}$ è polinomialmente riducibile a $\overline{SAT}$, implicando che $\overline{SAT}$ sia $\textsf{coNP}$-difficile per definizione. Dunque, segue la tesi.

Ho scritto la dimostrazione con $\overline{SAT}$ e non $UNSAT$ perché non è tecnicamente vero che $UNSAT := \overline{SAT}$, poiché $UNSAT$ è l'insieme delle formule booleane che non sono soddisfacibili da alcun assegnamento delle proprie variabili, mentre $\overline{SAT}$ contiene sia formule booleane non soddisfacibili, sia codifiche in input che non rappresentano formule booleane ben poste (ad esempio stringhe come $w = \land  \lor  x$); su internet c'è ambiguità a riguardo, perché alcune fonti le considerano uguali, altre no.

**Comment by Exyss** [comment]
Per il primo punto, ripropongo la soluzione presente sui miei appunti:

![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/bff812d7-d16a-4912-b293-20e69a694169)
![image](https://github.com/sapienzastudentsnetwork/acc2324/assets/65043207/1c291ec0-e78e-4905-9975-6dc0065afe9c)


-------

Per il secondo punto, la dimostrazione di @ph04 è corretta
